### PR TITLE
Support scheduling Cloud Agent on master nodes

### PR DESCRIPTION
**Pull request by DebakelOrakel**
## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

File: class/defaults.yml [hunk 1]
```diff
@@ -35,6 +35,8 @@ parameters:
       extraEnv: {}
       resourceRatio:
         memoryPerCore: 4Gi
+      nodeSelector: {}
+      tolerations: []
```
Comment (simu) [inline class/defaults.yml:39]: While you're at it, I'd make scheduling the agent on the master nodes the default:

```suggestion
      nodeSelector:
        node-role.kubernetes.io/master: ""
      tolerations:
        - effect: "NoSchedule"
          key: "node-role.kubernetes.io/master"
          operator: "Exists"
```

File: component/main.jsonnet [hunk 2]
```diff
@@ -8,23 +8,22 @@ local inv = kap.inventory();
...
-      SYNMonitoring: 'main',
-    };
+local namespaceLabels = (
+  if isOpenshift then {'openshift.io/cluster-monitoring': 'true'}
+  else {'SYNMonitoring': 'main'}
+) + params.namespaceLabels;
+local namespaceAnnotations = (
+  if isOpenshift then {'openshift.io/node-selector': ''}
+  else {}
+) + params.namespaceAnnotations;
```
Comment (simu) [inline component/main.jsonnet:18]: not sure why we even bother with openshift/non-openshift here, but doesn't really matter.

File: class/defaults.yml (hunk 1 above)
Comment (DebakelOrakel) [reply class/defaults.yml:39]: I thought about that, but that would make the component openshift specific which it is designed not to be.

**Comment by DebakelOrakel** [comment]
@simu Do you by chance have insight, what jsonnet linter is unhappy whit?

**Review by simu — APPROVED** [review]

**Pull request merged**